src/toffy/image_stitching.py

```diff
@@ -218,7 +265,81 @@ def stitch_images(
                 channels=[chan],
                 max_image_size=max_img_size,
             )
+            if img_size_scale:
+                image_data = rescale_stitched_array(image_data, img_size_scale)
+
             fname = os.path.join(stitched_subdir, chan + "_stitched.tiff")
             stitched = data_utils.stitch_images(image_data, num_cols)
-            current_img = stitched.loc["stitched_image", :, :, chan].values / scale
+            current_img = stitched.loc["stitched_image", :, :, chan].values / intensity_scale
             image_utils.save_image(fname, current_img)
+
+
+def rescale_image(img_data, scale, save_path=None):
+    """Rescale image data to a desired shape
+    Args:
+        img_data (np.array): data to be reshaped, expected to only be 2 dimensions
+        scale (int): amount to scale the data up or down
+        save_path (str): the location to save the tiff file
+    Returns:
+        numpy.array: data reshaped according to scale value
+    """
+    # check for 2d data
+    if len(img_data.shape) != 2:
+        raise ValueError("Image data must only have 2 dimensions.")
+
+    if scale < 1 and img_data.shape[0] % scale != 0:
+        raise ValueError("Scale value less than 1 must be a factor of the image size.")
+
+    # rescale data while preserving values
+    data_type = img_data.dtype
+    rescaled_data = transform.rescale(
+        np.array(img_data),
```

Comment:
No need to cast it as an array again.

Reply:
Fails without it bc of slicing an xarray in `rescale_stitched_array()`. But I'll move the cast up to that function to make sure the input is always an np.array.